Add referred facility to discharge serializer

Ashesh3 opened the pull request:
## Proposed Changes

- Add `referred_to` and `referred_to_external` to `PatientConsultationDischargeSerializer` to allow it to be set from the discharge form at the frontend.

### Associated PR

- Frontend: https://github.com/coronasafe/care_fe/pull/5730

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

sainak reviewed (commented):
@Ashesh3 a test is failing, also add a test for when `referred_to` is given and when both `referred_to` and `referred_to_external` are given

Ashesh3 commented:
> @Ashesh3 a test is failing, also add a test for when `referred_to` is given and when both `referred_to` and `referred_to_external` are given

Fixed and added more tests to cover `referred_to` specifically.
